src/os/windows/mod.rs

```diff
@@ -0,0 +1,287 @@
+                ASSOCSTR_COMMAND,
+                // In Shlwapi.h, this parameter's type is `LPCWSTR`.
+                // So it's not modified actually.
+                PWSTR(RUST_EXT.as_ptr() as _),
+                PWSTR::default(),
+                PWSTR(command.as_mut_ptr()),
+                &mut len as _,
+            )
+        }?;
+        return Ok(command);
+    }
+
+    // Replace %0 or %1 to arg1, and other % is unescape
+    fn replace_command_arg(arg: &OsStr, arg1: &OsStr) -> OsString {
+        let mut is_percent = false;
+        let mut iter = arg.encode_wide();
+        let mut buffer = vec![];
+        const ZERO: u16 = '0' as u16;
+        const ONE: u16 = '1' as u16;
+        const TWO: u16 = '2' as u16;
+        const NINE: u16 = '9' as u16;
+        const PERCENT: u16 = '%' as u16;
+        loop {
+            match (iter.next(), is_percent) {
+                (Some(ZERO..=ONE), true) => {
+                    buffer.extend(arg1.encode_wide());
+                }
+                (Some(TWO..=NINE), true) => {
+                    // Nothing to do.
+                }
+                (Some(PERCENT), false) => {
+                    is_percent = true;
+                    continue;
+                }
+                (Some(c), _) => {
+                    buffer.push(c);
+                }
+                (None, _) => break,
+            }
+            is_percent = false;
+        }
+        OsString::from_wide(&buffer)
+    }
+}
+
+pub fn open_file_with(
+    application: impl AsRef<OsStr>,
+    path: impl AsRef<OsStr>,
+) -> Result<(), OpenFileError> {
+    // In windows, there is no standerd way to find application by name.
+    match application.as_ref().to_str() {
+        Some("Android Studio") => open_file_with_android_studio(path),
+        _ => {
+            unimplemented!()
+        }
+    }
+}
+
+const ANDROID_STUDIO_UNINSTALL_KEY_PATH: &[u16] = const_utf16::encode_null_terminated!(
+    "SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\Android Studio"
+);
+const ANDROID_STUDIO_UNINSTALLER_VALUE: &[u16] =
+    const_utf16::encode_null_terminated!("UninstallString");
+#[cfg(target_pointer_width = "64")]
+const STUDIO_EXE_PATH: &str = "bin/studio64.exe";
+#[cfg(target_pointer_width = "32")]
+const STUDIO_EXE_PATH: &str = "bin/studio.exe";
+
+fn open_file_with_android_studio(path: impl AsRef<OsStr>) -> Result<(), OpenFileError> {
+    let mut buffer = [0; MAX_PATH as usize];
+    let lstatus = unsafe {
+        SHRegGetPathW(
+            HKEY_LOCAL_MACHINE,
+            PWSTR(ANDROID_STUDIO_UNINSTALL_KEY_PATH.as_ptr() as _),
+            PWSTR(ANDROID_STUDIO_UNINSTALLER_VALUE.as_ptr() as _),
+            PWSTR(buffer.as_mut_ptr()),
+            0,
+        )
+    };
+    if lstatus.0 as u32 != ERROR_SUCCESS.0 {
+        return Err(OpenFileError::OSError(windows::Error::from_win32()));
+    }
+    let len = NullTerminatedWTF16Iterator(buffer.as_slice().as_ptr()).count();
```

Comment:
`as_slice()` on arrays is a nightly feature, you should probably use `&buffer[..]`

Reply:
Oh, no. I didn't need to make it a slice, so I deleted it.